### DISCUSSION TITLE
Customisable warning messages

### DISCUSSION THREAD

**Discussion by ritz078** [adobe/react-spectrum#4768] [Ideas]
I am using `useCheckbox` and I am getting the error message as `If you do not provide children, you must specify an aria-label for accessibility`. This is great but the problem is that instead of using children as label, we are using a `label` prop which makes this error message a bit confusing. Is it possible to customise this error message somehow? If not, this will be a great addition to the library as not everyone might be using the same prop to achieve the same thing.

**Comment by LFDanLu** [comment]
What did you have in mind for error message customization? I'm concerned that full customization of the message could be abused and it doesn't really seem to have a use case outside of this specific kind of case. Could you instead provide `label` as the `children` prop of the hook to silence the warning instead?

**Reply by ritz078** [reply]
> What did you have in mind for error message customization?

"If you do not provide **label**, you must specify an aria-label for accessibility"

> Could you instead provide label as the children prop of the hook to silence the warning instead?

I am passing label as children right now to hook but if label is undefined, the error thrown is about children being undefined because for hooks, it's called children. What if you just make the prop name customisable ? Assuming that labels are always passed as children might be too generic ?

**Reply by LFDanLu** [reply]
I feel like making the prop name customizable just for the warning may be a bit overkill, but I do see you point in it being confusing since the console warning doesn't point directly back to useToggle where the `children` context would make sense. Would making the message mention the hook where the warning lives be more clear? 

e.g. "If you do not provide children to useToggle, you must specify an aria-label for accessibility"

**Reply by ritz078** [reply]
The problem with using the hook name is that for the user using the component, it doesn't make any sense as it's just an internal implementation. They might not even be aware what I mean by `useToggle`. The error should instead mention something from the public API which isn't the hook. 

For example: If I am using react-aria-components and I see a warning about useToggleButton then I will confuse me even further. The user passed the value to the component and isn't aware of the existence of useToggleButton.

Does it make sense?

One alternative might be to provide an option to disable that warning so that I can handle that my own but I don't see this being a better alternative than allowing customisation.
